### PR TITLE
T1441

### PR DESCRIPTION
**Pull request by ScottDowne**

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @humphd received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/9bc6a652aa484e2/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @humphd received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/3c086e40e638e2e/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/9bc6a652aa484e2/output.txt

Total script time: 0.54 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/3c086e40e638e2e/output.txt

Total script time: 0.53 mins

Server is running on http://pr473.butyr.org/templates/test.html

File: css/butter.ui.deprecated.css [hunk 1]
```diff
@@ -2,6 +2,30 @@
  * If a copy of the MIT license was not distributed with this file, you can
  * obtain one at http://www.mozillapopcorn.org/butter-license.txt */
 
+@font-face {
+  font-family: 'Open Sans';
+  font-style: italic;
+  font-weight: 400;
+  src: local('Open Sans Italic'), local('Open-Sans-Italic'),
+    url('../resources/fonts/OpenSans-Italic.ttf') format('truetype');
```
Comment (humphd) [inline css/butter.ui.deprecated.css:10]: In other cases we've favoured embedding a datauri so the images are inline and all you need is the css file.  I think for 0.5.x this is fine, but if you're thinking of landing in 0.6, we should get a ticket to fix that on master.  Same with the ones below.

File: dialogs/view-source.html [hunk 2]
```diff
@@ -0,0 +1,69 @@
+<!-- This Source Code Form is subject to the terms of the MIT license
+If a copy of the MIT license was not distributed with this file, you can
+obtain one at http://www.mozillapopcorn.org/butter-license.txt -->
+
+<html>
+  <head>
+    <title>Export</title>
+    <script type="text/javascript" src="dialog.js"></script>
+    <script type="text/javascript">
```
Comment (humphd) [inline dialogs/view-source.html]: Move this to an external .js file so we can lint it.

File: dialogs/view-source.html [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+          htmlExport.style.display = "block";
+          jsonExport.style.display = "none";
+        }, false );
+
+        Dialog.wait( "export", function( e ){
+          try{
+            jsonExport.value = JSON.stringify( e.data.json, null, 2 );
+          }
+          catch( e ){
+            jsonExport.value = "There was an error trying to parse the JSON blob for this project. Please file a bug at https://webmademovies.lighthouseapp.com/projects/65733-butter/ and let us know.";
```
Comment (humphd) [inline dialogs/view-source.html]: s/blob/file/ ?

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @ScottDowne received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/0f08319b07152c3/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/0f08319b07152c3/output.txt

Total script time: 0.80 mins

File: dialogs/view-source.html (hunk 3 above)
Comment (ScottDowne) [inline dialogs/view-source.html]: Maybe just remove blob and add nothing?

Comment (humphd) [inline dialogs/view-source.html]: I just think "blob" is a technical term that sounds like it's made-up to a user.  JSON or JSON file are both fine.

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @ScottDowne received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/8f827edc3a44320/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @ScottDowne received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/78f47fe7830a60e/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/8f827edc3a44320/output.txt

Total script time: 0.31 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/78f47fe7830a60e/output.txt

Total script time: 0.61 mins

Server is running on http://pr473.butyr.org/templates/test.html

File: src/cornfield/module.js [hunk 4]
```diff
@@ -125,15 +125,20 @@ define(['util/xhr'], function(XHR) {
       XHR.get(server + "/browserid/logout", function() {
         email = null;
         if (this.readyState === 4) {
+          var response;
+
           try {
-            var response = JSON.parse(this.response);
+            response = JSON.parse( this.response );
             authenticated = false;
             email = undefined;
```
Comment (humphd) [inline src/cornfield/module.js]: don't set these to undefined, use null, unless we have code that depends on undefined?  Same with ones below.

File: css/butter.ui.deprecated.css [hunk 5]
```diff
@@ -268,6 +292,7 @@ input.form-error {
   background-image: -moz-linear-gradient( center bottom, #2C2C2C 0, #4C4C4C 100% );
   background-image: -ms-linear-gradient( center bottom, #2C2C2C 0, #4C4C4C 100% );
   display: block;
+  font-family: "Open Sans";
```
Comment (k88hudson) [inline css/butter.ui.deprecated.css]: This needs to be font-family: "Open Sans", "Helvetica Neue", sans-seirf  --- we don't want the fallback to be Times.

File: src/cornfield/module.js (hunk 4 above)
Comment (ScottDowne) [inline src/cornfield/module.js]: They are expected to be strings, not objects. Should we set them to empty strings?

Going to try that and run it through the tests.

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @ScottDowne received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/d436ef0c2e2f67b/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @ScottDowne received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/5240a1caf295917/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/d436ef0c2e2f67b/output.txt

Total script time: 0.99 mins

Server is running on http://pr473.butyr.org/templates/test.html

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/5240a1caf295917/output.txt

Total script time: 0.30 mins

**Comment by benrito** [comment]
Add or change these lines and viola. Shoot it back and I will PR+

div.editor-actions {
  margin-top: 10px;
  margin-right: 3px
}

butter-header {
    50px;
}

butter-header-spacing {
    50px

}

butter-drop {
    url: url("../resources/minilogo.png") <--- in your inbox
    width: 97px
    height: 40px
}

#butter-header h1
    left: 40%
    top: 14px;
}

**Pull request merged**